src/transformers/models/mixtral/modeling_mixtral.py

```diff
@@ -1190,6 +1192,7 @@ def forward(
                 (batch_size, seq_length),
                 inputs_embeds,
                 past_key_values_length,
+                sliding_window=self.config.sliding_window if is_torch_version_greater_or_equal_than_2_2_0 else None,
```

Comment:
The issue here is that `_prepare_4d_causal_attention_mask_for_sdpa` seems to return None if `attention_mask` is `None` (which is the case in the test) while if we actually want to use sliding we need to return the full causal mask. cc @fxmarty